Lib/pty.py

```diff
@@ -151,20 +161,23 @@ def spawn(argv, master_read=_read, stdin_read=_read):
     """Create a spawned process."""
     if type(argv) == type(''):
         argv = (argv,)
+
     pid, master_fd = fork()
     if pid == CHILD:
         os.execlp(argv[0], *argv)
+
     try:
-        mode = tty.tcgetattr(STDIN_FILENO)
-        tty.setraw(STDIN_FILENO)
-        restore = 1
+        mode = tcgetattr(STDIN_FILENO)
+        setraw(STDIN_FILENO)
+        restore = True
     except tty.error:    # This is the same as termios.error
-        restore = 0
+        restore = False
+
     try:
         _copy(master_fd, master_read, stdin_read)
-    except OSError:
```

Comment:
Here  I will made some checke if _copy work correctly

Reply:
I'm not sure I follow what you're thinking.  Assuming we return a boolean value from `_copy_ as you suggest, what should happen if `_copy` returns `False?  I don't think there's any code to be run only under one half of that conditional...

Reply:
Note to self: this replaces `except OSError` because `_copy` handles that case internally now. Remaining possible OSErrors from other cases are now propagated to the user which is an improvement.